transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollStreamChannel.java

```diff
@@ -824,7 +824,7 @@ void epollInReady() {
                     allocHandle.incMessagesRead(1);
                     pipeline.fireChannelRead(byteBuf);
                     byteBuf = null;
-                } while (allocHandle.continueReading());
+                } while (config.isAutoRead() && allocHandle.continueReading());
```

Comment:
I agree that the logic for determining that the read loop should exit should live in the ```RecvByteBufAllocator.DelegatingHandle```.  This is just a very rough draft to prove what might allow us to provide *normal behaviour* for autoRead in EDGE mode.

The ```RecvByteBufAllocator.DelegatingHandle``` Epoll implementations will completely disregard the value of ```config.isAutoRead()``` when in EDGE mode (because of the logic [here](https://github.com/netty/netty/blob/4.1/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollRecvByteAllocatorMessageHandle.java#L37) and [here](https://github.com/netty/netty/blob/4.1/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollRecvByteAllocatorStreamingHandle.java#L39)). So to get around that, I added the ```config.isAutoRead()``` check to the read loop.  This way if an application's ```ChannelHandler``` sets ```autoRead``` to ```false``` while executing ```channelRead``` the transport (in this specific case, EpollMode.ET) will honour that and break out of the loop.

Again I know this is not the right place to put the check, the check should be in the ```RecvByteBufAllocator``` implementations.  I would like to get @normanmaurer to give me some tips on the best way to accomplish that.